src/common/config/in-memory-configuration.ts

```diff
@@ -128,15 +128,24 @@ export class InMemoryConfiguration implements Configuration {
   @Transform(({ value }) => parseInt(value, 10), { toClassOnly: true })
   REGISTRY_KEYS_QUERY_CONCURRENCY = 5;
 
+  @ValidateIf((conf) => !conf.KEYS_API_URL)
   @IsNotEmpty()
   @IsNumber()
   @Min(1)
   @Transform(({ value }) => parseInt(value, 10), { toClassOnly: true })
-  KEYS_API_PORT = 3001;
+  KEYS_API_PORT = 0;
 
-  @IsOptional()
+  @ValidateIf((conf) => !conf.KEYS_API_URL)
+  @IsNotEmpty()
+  @IsString()
+  KEYS_API_HOST = '';
+
+  @ValidateIf((conf) => {
+    return !conf.KEYS_API_PORT && !conf.KEYS_API_HOST;
```

Comment:
so we want to use   KEYS_API_URL only if both conf.KEYS_API_PORT  and conf.KEYS_API_HOST were not set, right ?